src/test/coins_tests.cpp

```diff
@@ -589,314 +632,245 @@ static void SetCoinsValue(CAmount value, Coin& coin)
     }
 }
 
-static size_t InsertCoinsMapEntry(CCoinsMap& map, CoinsCachePair& sentinel, CAmount value, char flags)
+static size_t InsertCoinsMapEntry(CCoinsMap& map, CoinsCachePair& sentinel, const CoinStruct& cache_coin)
 {
-    if (value == ABSENT) {
-        assert(flags == NO_ENTRY);
-        return 0;
-    }
-    assert(flags != NO_ENTRY);
     CCoinsCacheEntry entry;
-    SetCoinsValue(value, entry.coin);
-    auto inserted = map.emplace(OUTPOINT, std::move(entry));
-    assert(inserted.second);
-    inserted.first->second.AddFlags(flags, *inserted.first, sentinel);
-    return inserted.first->second.coin.DynamicMemoryUsage();
+    SetCoinsValue(cache_coin.value, entry.coin);
+    auto [iter, inserted] = map.emplace(OUTPOINT, std::move(entry));
+    assert(inserted);
+    if (cache_coin.IsDirty()) iter->second.SetDirty(*iter, sentinel);
+    if (cache_coin.IsFresh()) iter->second.SetFresh(*iter, sentinel);
+    return iter->second.coin.DynamicMemoryUsage();
 }
 
-void GetCoinsMapEntry(const CCoinsMap& map, CAmount& value, char& flags, const COutPoint& outp = OUTPOINT)
+MaybeCoin GetCoinsMapEntry(const CCoinsMap& map, const COutPoint& outp = OUTPOINT)
 {
-    auto it = map.find(outp);
-    if (it == map.end()) {
-        value = ABSENT;
-        flags = NO_ENTRY;
-    } else {
-        if (it->second.coin.IsSpent()) {
-            value = SPENT;
-        } else {
-            value = it->second.coin.out.nValue;
-        }
-        flags = it->second.GetFlags();
-        assert(flags != NO_ENTRY);
+    if (auto it{map.find(outp)}; it != map.end()) {
+        return CoinStruct{
+            it->second.coin.IsSpent() ? SPENT : it->second.coin.out.nValue,
+            ToCaching(it->second.IsDirty(), it->second.IsFresh())};
```

Comment:
Do we need a new function if we're only calling it the one time here? Just seems like unnecessary misdirection.

Reply:
Converting flags to enums isn't the responsibility of this method, that's why I've extracted it.